frontend/public/components/overview/pods-overview.tsx

```diff
@@ -123,25 +132,49 @@ export const PodsOverview: React.SFC<PodsOverviewProps> = ({
     metadata: { name, namespace },
   } = obj;
 
+  const [showWaitingPods, setShowWaitingPods] = React.useState(false);
+  const showWaitingForBuildAlert =
+    isDeploymentGeneratedByWebConsole(obj) && pods.some(isPodWithoutImageId);
+
+  let filteredPods = [...pods];
+  if (showWaitingForBuildAlert && !showWaitingPods) {
+    filteredPods = filteredPods.filter((pod) => !isPodWithoutImageId(pod));
+  }
+  filteredPods.sort(podCompare);
+
   const errorPodCount = _.size(_.filter(pods, (pod) => isPodError(pod)));
   const podsShown = Math.max(Math.min(errorPodCount, MAX_ERROR_PODS), MAX_PODS);
   const linkTo = allPodsLink || `${resourcePath(referenceFor(obj), name, namespace)}/pods`;
   const emptyMessage = emptyText || 'No Pods found for this resource.';
-  pods.sort(podCompare);
 
   return (
     <>
       <SidebarSectionHeading text="Pods">
-        {_.size(pods) > podsShown && (
+        {_.size(filteredPods) > podsShown && (
           <Link className="sidebar__section-view-all" to={linkTo}>
-            {`View all (${_.size(pods)})`}
+            {`View all (${_.size(filteredPods)})`}
```

Comment:
Small nit: I think this should still show the size for `pods`. Because when you go to the details page, you would see the list of all pods, including the ones that aren't filtered. Not entirely sure though, WDYT?

Reply:
I think you're right here, will change it back. Thanks Saud :+1:

Reply:
Done